src/main/java/org/terasology/signalling/componentSystem/SignalSwitchBehaviourSystem.java

```diff
@@ -360,8 +360,8 @@ private void handlePressurePlateEvents() {
 
         Iterable<EntityRef> players = entityManager.getEntitiesWith(CharacterComponent.class, LocationComponent.class);
         for (EntityRef player : players) {
-            Vector3f playerLocation = player.getComponent(LocationComponent.class).getWorldPosition();
-            Vector3i locationBeneathPlayer = new Vector3i(playerLocation.x + 0.5f, playerLocation.y - 0.5f, playerLocation.z + 0.5f);
+            Vector3f playerLocation = player.getComponent(LocationComponent.class).getWorldPosition(new Vector3f());
+            Vector3i locationBeneathPlayer = new Vector3i(new Vector3f(playerLocation.x + 0.5f, playerLocation.y - 0.5f, playerLocation.z + 0.5f), RoundingMode.FLOOR);
```

Comment:
as `playerLocation` is only used right here, maybe we can make better use of vector arithmetics:

```suggestion
            Vector3f playerLocation = player.getComponent(LocationComponent.class).getWorldPosition(new Vector3f());
            Vector3i locationBeneathPlayer = new Vector3i(playerLocation.add(0.5f, -0.5f, 0.5f, RoundingMode.FLOOR);
```